### PR TITLE
Drop references to auth-path and kubernetes_auth

### PR DESCRIPTION
**Pull request by skitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

clientcmd still has a few references to `auth-path` and  `~/.kubernetes_auth`. These are long gone, removed in 2016:
* AuthPath removal: https://github.com/kubernetes/kubernetes/pull/29216
* Flag removal: https://github.com/kubernetes/kubernetes/pull/40048

This removes the remnants from clientcmd, mostly in the comments
describing how the configuration is loaded.

Since getServerIdentificationPartialConfig can no longer fail (it
copies fields from one struct to another), this drops the error
return, along with the error handling in the caller.

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by cici37** [comment]
Hi @skitt Thanks for raising the PR! Would you mind sharing more info on the background of the task or any issue related? Thanks :)

**Comment by fedebongio** [comment]
Made a pass on the triage today, waiting for answer of author

**Comment by skitt** [comment]
@cici37 I’ve updated the PR description and commit message to add more information. I came across this when digging into `clientcmd`; there’s no particular background to the PR, it’s just cleaning up stuff I found confusing.

**Comment by cici37** [comment]
/sig auth
for visibility

**Review by stlaz — COMMENTED** [review]
/lgtm

nit comment, feel free to ignore if you want

File: staging/src/k8s.io/client-go/tools/clientcmd/client_config.go [hunk 1]
```diff
@@ -243,43 +243,32 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
...
-// 1.  configClusterInfo (the final result of command line flags and merged .kubeconfig files)
-// 2.  configAuthInfo.auth-path (this file can contain information that conflicts with #1, and we want #1 to win the priority)
-// 3.  load the ~/.kubernetes_auth file as a default
-func getServerIdentificationPartialConfig(configAuthInfo clientcmdapi.AuthInfo, configClusterInfo clientcmdapi.Cluster) (*restclient.Config, error) {
-	mergedConfig := &restclient.Config{}
-
-	// configClusterInfo holds the information identify the server provided by .kubeconfig
+// both, so we have to split the objects and merge them separately.
+// configClusterInfo (the final result of command line flags and merged .kubeconfig files) is now the only source of
+// service identification.
```
[the remark this answers is not in the document]
Comment (skitt) [reply staging/src/k8s.io/client-go/tools/clientcmd/client_config.go]: Ah yes, I like that better. The introductory comment is the same for both functions so I’ve separated it from the function docs, and I’ve updated `getUserIdentificationPartialConfig`’s description to follow the same pattern:

```
// getUserIdentificationPartialConfig extracts user identification information from configAuthInfo
// (the final result of command line flags and merged .kubeconfig files);
// if the information available there is insufficient, it prompts (if possible) for additional information.
```

**Comment by stlaz** [comment]
/lgtm
Thank you for addressing the nit, the change looks good to me :+1:

**Pull request merged**